lib/rbs/prototype/rb.rb

```diff
@@ -195,6 +196,11 @@ def process(node, decls:, comments:, context:)
 
           decls.push member unless decls.include?(member)
 
+          new_ctx = Context.initial.tap { |ctx| ctx.singleton = kind == :singleton }
```

Comment:
```suggestion
          new_ctx = context.dup.tap { |ctx| ctx.singleton = kind == :singleton }
```

Because the new context should inherit the namespace context.
Currently it does not use the namespace context inside the method definition. But the new context should keep the parent context's namespace for the future. 


---


BTW, I'd like to refactor the context class to implement more concrete methods to update the context. Currently it just provides primitive interfaces and it introduces confusion.